### PR TITLE
Delete field plans by holding left mouse button

### PR DESCRIPTION
**Pull request by reyandme**

### CODE REVIEW

File: src/KM_Defaults.pas [hunk 1]
```diff
@@ -260,6 +260,14 @@ interface
     cmMarkers //CenterScreen, Defence, FOW markers
     );
 
+type
+  TKMCursorFieldMode = (
```
Comment (Kromster80) [inline src/KM_Defaults.pas:264]: Comments would help here, explaining what this enum type represents

File: src/gui/KM_InterfaceGamePlay.pas [hunk 2]
```diff
@@ -2822,6 +2850,19 @@ procedure TKMGamePlayInterface.MouseDown(Button: TMouseButton; Shift: TShiftStat
 // 2. Perform SelectingTroopDirection if it is active
 // 3. Display various cursors depending on whats below (might be called often)
 procedure TKMGamePlayInterface.MouseMove(Shift: TShiftState; X,Y: Integer);
+  procedure HandleFieldLMBDrag(P: TKMPoint; aFieldType: TFieldType);
+  begin
+    if not KMSamePoint(LastDragPoint, P) then
+      if (gMySpectator.Hand.CanAddFakeFieldPlan(P, aFieldType)) and (gGameCursor.Tag1 = Byte(cfmPlan)) then
```
Comment (Kromster80) [inline src/gui/KM_InterfaceGamePlay.pas:2856]: This is largely my mistake in the past. `Byte(something)` should be really `Ord(something)` throughout the code.

File: src/hands/KM_Hand.pas [hunk 3]
```diff
@@ -472,37 +474,46 @@ procedure TKMHand.AddField(aLoc: TKMPoint; aFieldType: TFieldType);
 end;
 
 
+function TKMHand.LocHasNoAllyPlans(aLoc: TKMPoint): Boolean;
+var I: Byte;
```
Comment (Kromster80) [inline src/hands/KM_Hand.pas:478]: Change Byte to Integer please, as discussed in other PR.

**Comment by reyandme** [comment]
Find differ in behaviour with current version - now if u start placing road in the fog of war, then drag out of it - u will able to place road (only where FOW is reaveled).
after this feature pull request you are not able to do that, so you have to start from tiles, where FOW is revealed.
I think need to fix this.

**Comment by reyandme** [comment]
Also I think we need to rename gic_BuildAddFieldPlan to gic_BuildToggleFieldPlan, because we actually toggle field plan state, but not add. Name gic_BuildAddFieldPlan confused me at the first moment.

File: src/gui/KM_InterfaceGamePlay.pas (hunk 2 above)
Comment (reyandme) [reply src/gui/KM_InterfaceGamePlay.pas:2856]: looks same here. But I can change to Ord here, as well as in other lines in this file and in RenderPool

**Comment by reyandme** [comment]
To avoid multi commits as in another PR I will squash commits in local rep first, then publish polished version.

**Comment by Kromster80** [comment]
Merged through TortoiseGit (applied as patch)

**Pull request closed without merging**